transaction/core/src/account_keys.rs

```diff
@@ -360,10 +363,27 @@ impl AccountKey {
             fog_authority_sig: Default::default(),
         };
 
-        // Compute fog_authority_sig as a signature over self.fog_authority_key_fingerprint
+        // NOTE: The fog_authority_sig is deterministic due to using the private key hash as the rng seed
         if !self.fog_report_url.is_empty() {
-            // FIXME: FOG-106 fog_authority_sig should be a Schnorrkel sig using subaddress_view_private
-            result.fog_authority_sig.extend(&[9u8, 9u8, 9u8, 9u8]);
+            // Construct the fog authority signature over the fingerprint using the view privkey
+            let scalar: Scalar = self.subaddress_view_private(index).scalar();
+
+            // Create rng seed from hash of private key in order to create nonce and sign
+            let seed: [u8; 32] = Sha256::digest(&scalar.to_bytes()).into();
+            let mut cspring: FixedRng = SeedableRng::from_seed(seed);
+            let mut nonce = [0u8; 32];
+            cspring.fill_bytes(&mut nonce);
+
+            let mut secret_bytes = [0u8; 64];
+            secret_bytes[0..32].copy_from_slice(&scalar.to_bytes());
+            secret_bytes[32..64].copy_from_slice(&nonce);
+
+            let secret_key = SecretKey::from_bytes(&secret_bytes).unwrap();
```

Comment:
Why are we not using the mini secret key, and letting them do the nonce generation?

Reply:
The MiniSecretKey expansion creates a new KeyPair that does not verify with the RistrettoPublic associated with the scalar. That's what [MC-1611](https://mobilecoin.atlassian.net/browse/MC-1611) should address, so that our keys are doing the same thing that Schnorrkel keys are doing.

Reply:
Similar discussion here: https://github.com/mobilecoinofficial/mobilecoin/pull/246#discussion_r448661514